_posts/2022-XX-XX-configuring-git-repositories.adoc

```diff
@@ -0,0 +1,267 @@
+
+* Enter the {prod-short} OAuth callback URL as the *Redirect URI*. The default value is \https://{prod-deployment}-{prod-namespace}.__<domain>__/api/oauth/callback.
+
+* Check the *Confidential* and *Expire* access tokens checkboxes.
+
+* Under *Scopes*, check the `api`, `write_repository`, and `openid` checkboxes.
+
+Click *Save application* to create the Authorized Application.
+
+Copy and save the *Application ID* and *Secret* values because you will provide them to your {prod-short} administrator.
+
+===== Creating the Bitbucket Application Link
+
+
+Create an RSA key-value pair:
+Use a command line to generate the *public-stripped.pub* and *privatepkcs8-stripped.pem* keys in the required format:
+----
+$ openssl genrsa -out private.pem 2048
+
+$ openssl rsa -in private.pem -pubout > public.pub
+
+$ openssl pkcs8 -topk8 -inform pem -outform pem -nocrypt -in private.pem -out privatepkcs8.pem
+
+$ cat public.pub | sed 's/-----BEGIN PUBLIC KEY-----//g' | sed 's/-----END PUBLIC KEY-----//g' | tr -d '\n' > public-stripped.pub
+
+$ cat privatepkcs8.pem | sed 's/-----BEGIN PRIVATE KEY-----//g' |  sed 's/-----END PRIVATE KEY-----//g' | tr -d '\n' > privatepkcs8-stripped.pem
+----
+
+
+Use a command line to generate a consumer key and a shared secret:
+
+----
+$ openssl rand -base64 24 > bitbucket-consumer-key
+
+$ openssl rand -base64 24 > bitbucket-shared-secret
+----
+
+Create the *Application* link:
+
+Enter the {prod-short} URL as the *Application URL*. The default value is `\https://che-eclipse-che.__<domain>__/ ` .
+
+Click *Create new link*.
+
+Under *The supplied Application URL has redirected once*, check the *Use this URL* checkbox and click *Continue*.
+
+Enter the following details for the *Application* link:
+
+* Enter *{prod-short}* as the *Application Name*.
+
+* Select *Generic Application* as the *Application Type*.
+
+* Enter *{prod-short}* as the *Service Provider Name*.
+
+* Paste the content of the *bitbucket-consumer-key* file as the *Consumer key*.
+
+* Paste the content of the *bitbucket-shared-secret* file as the *Shared secret*.
+
+* Enter `__<bitbucket_server_url>__/plugins/servlet/oauth/request-token` as the *Request Token URL*.
+
+* Enter `__<bitbucket_server_url>__/plugins/servlet/oauth/access-token` as the *Access token URL*.
+
+* Enter `__<bitbucket_server_url>__/plugins/servlet/oauth/access-token` as the *Authorize URL*.
+
+* Check the *Create incoming link* checkbox.
+
+Click *Continue* and enter the following details:
+
+* Paste the content of the *bitbucket_consumer_key* file as the *Consumer Key*.
+
+* Enter *{prod-short}* as the *Consumer name*.
+
+* Paste the content of the *public-stripped.pub* file as the *Public Key*.
+
+Click *Continue*.
+
+This completes your user steps. Additional steps are now required from your {prod-short} administrator.
+
+
+==== Admin side
+
+After you've created the reference on the Git SCM, your {prod-short} administrator will create the following Kubernetes secret.
```

Comment:
Let's say I am a Che user. Can I have my own private fork of our team's private project repository. If I set up, let's say a GitHub OAuth App, and the Che admin applies a Secret for it. Will this Secret enable my colleagues to access my private (my fork) repository?

Reply:
You need to go through the oauth flow to get an access token (which is then used to read/write etc). This access token is created for your Che user in his k8s namespace. If your colleague wanted his own access token, he would need to go through the oauth flow as well. Which means he would need to login to GitHub with your GitHub credentials.